### PR TITLE
Enable JGroups encryption via CLI operations

### PR DESCRIPTION
**Pull request by yersan**
Enable JGroups encryption via CLI operations. Behave and bats tets passing.

It requires: https://github.com/wildfly/temp-eap-modules/pull/42

### CODE REVIEW

File: jboss/container/wildfly/launch/jgroups/added/launch/ha.sh [hunk 1]
```diff
@@ -196,7 +186,7 @@ generate_dns_ping_config() {
     fi
 
     if [ "${ping_protocol}" = "openshift.KUBE_PING" ]; then
-        ping_protocol="kubernetes.KUBE_PING" # openshift.KUBE_PING is deprecated and removed, but aliased
+        ping_protocol=" .KUBE_PING" # openshift.KUBE_PING is deprecated and removed, but aliased
```
Comment (jfdenise) [inline jboss/container/wildfly/launch/jgroups/added/launch/ha.sh]: Is it expected to prefix the protocol with a white space?

File: jboss/container/wildfly/launch/jgroups/added/launch/jgroups.sh [hunk 2]
```diff
@@ -43,32 +76,133 @@ create_jgroups_encrypt_asym() {
...
+        while read -r protocolType; do
+          if [ ${protocolType} == "pbcast.NAKACK2" ]; then
+            break
+          fi
+          ((index=index+1))
+        done <<< "${protocolTypes}"
+
+        if [ ${index} -eq ${#arr[@]} ]; then
+          echo "You have set JGROUPS_CLUSTER_PASSWORD environment variable to configure ASYM_ENCRYPT protocol but pbcast.NAKACK2 protocol was not found for ${stack^^} stack. Fix your configuration to contain the pbcast.NAKACK2 in the JGroups subsystem for this to happen." >> "${CLI_SCRIPT_ERROR_FILE}"
+          missingNAKACK2="true"
```
Comment (jfdenise) [inline jboss/container/wildfly/launch/jgroups/added/launch/jgroups.sh]: You should "continue", right?

File: jboss/container/wildfly/launch/jgroups/added/launch/jgroups.sh [hunk 3]
```diff
@@ -43,32 +76,133 @@ create_jgroups_encrypt_asym() {
...
+      while read -r protocolType; do
+        if [ "${protocolType}" = "pbcast.NAKACK2" ]; then
+          break
+        fi
+        ((index=index+1))
+      done <<< "${protocolTypes}"
+     
+      if [ ${index} -eq ${#arr[@]} ]; then
+        echo "You have set JGROUPS_CLUSTER_PASSWORD environment variable to configure SYM_ENCRYPT protocol but pbcast.NAKACK2 protocol was not found for ${stack^^} stack. Fix your configuration to contain the pbcast.NAKACK2 in the JGroups subsystem for this to happen." >> "${CLI_SCRIPT_ERROR_FILE}"
+        missingNAKACK2="true"
```
Comment (jfdenise) [inline jboss/container/wildfly/launch/jgroups/added/launch/jgroups.sh]: You should "continue", right?

File: jboss/container/wildfly/launch/jgroups/added/launch/jgroups.sh [hunk 4]
```diff
@@ -99,40 +233,93 @@ validate_keystore_legacy() {
...
-            jgroups_encrypt=$(create_jgroups_elytron_legacy "${JGROUPS_ENCRYPT_KEYSTORE}" "${JGROUPS_ENCRYPT_PASSWORD}" "${JGROUPS_ENCRYPT_NAME}" "${JGROUPS_ENCRYPT_KEYSTORE_DIR}")
-        fi
-    elif [ "${keystore_validation_state}" = "partial" ]; then
-        keystore_warning_message="${jgroups_unencrypted_message//<STATE>/partial}"
+    if [ "${has_elytron_subsystem}" -eq 0 ]; then
+      if [ "${key_store_conf_mode}" = "xml" ]; then
+        validate_keystore_and_create "xml"
+      elif [ "${key_store_conf_mode}" = "cli" ]; then
+        validate_keystore_and_create "cli"
+      elif [ "${CONFIG_ADJUSTMENT_MODE,,}" = "xml" ]; then
```
Comment (jfdenise) [inline jboss/container/wildfly/launch/jgroups/added/launch/jgroups.sh]: There is a lot of context...What is the difference with"${ key_store_conf_mode}" = "xml". Is that is obvious, forget my question...

Comment (yersan) [reply jboss/container/wildfly/launch/jgroups/added/launch/jgroups.sh]: This is here to cover the following case: User has strictly defined that he wants only replacement via xml markers, we have the Elytron subsystem but we do not have the Keystore marker, in that case, then replace the protocol using the "legacy" style, which means include the key store in the JGroups protocol itself.

Comment (yersan) [reply jboss/container/wildfly/launch/jgroups/added/launch/jgroups.sh]: Not so obvious indeed

Comment (jfdenise) [reply jboss/container/wildfly/launch/jgroups/added/launch/jgroups.sh]: Thank-you, perhaps a comment would help?

Comment (yersan) [reply jboss/container/wildfly/launch/jgroups/added/launch/jgroups.sh]: Yes, done

**Pull request merged**